### PR TITLE
fuelup mdbook

### LINKED ISSUE #3: Add a small `mdbook` for `fuelup`
Similar to how [`rustup` has an mdbook](https://rust-lang.github.io/rustup/).

It could be worth starting this early so that we can document features as they're added.

We can take inspiration from `sway`, `fuels-rs` or `fuel-indexer` repos on how to setup an mdbook and how to automate it's publishing with CI.

### LINKED ISSUE #112: Add CONTRIBUTING.md
Add a CONTRIBUTING.md document for raising issues and/or contributing code.

### LINKED ISSUE #118: Include a markdown table in `CHANNELS.md` or `README.md` as a high-level overview of the available channels
Perhaps it could be useful to have something like this in `CHANNELS.md` or even the `README.md`?

| Channel | Source | Integration Tested | Update Frequency | Available |
| --- | --- | --- | --- | --- |
| **latest** | Published bins. | :heavy_check_mark: | Checks every 30 min. | :heavy_check_mark: |
| **nightly** | `master` branch. | :heavy_check_mark: | Nightly. | :heavy_minus_sign: |
| **beta** | Published bins. | :heavy_check_mark: | 6 Week Cycle. | :heavy_minus_sign: |
| **stable** | Published bins. | :heavy_check_mark: + successful beta | 6 Week Cycle. | :heavy_minus_sign: |

Could update these in the future to use badges that link to the latest github workflow for each, and maybe provide another column that shows the current versions released under each channel.

### PR DESCRIPTION
**Pull request by bingcicle**
closes #3 (finally)
closes #112 
closes #118 

I will do another read-through again just to proofread this before I mark this as ready for review because I wrote all of these in one go but this should ideally go in with the v0.3.0 release.

### CODE REVIEW

**Comment by bingcicle** [comment]
I've included the Channels information indicated in #118. Are those timelines a lock-in at this point or should we perhaps leave some of the timeline blank?

**Comment by bingcicle** [comment]
I think the content of this book should be ready for review, as well as the github workflow to publish the book.

One note is that I was initially referring to the toolchain as the Fuel toolchain; but as I was cross referring to the Sway book, we call it the Sway toolchain there instead. I decided to follow that convention for consistency.

File: docs/src/concepts/channels/latest.md [hunk 1]
```diff
@@ -1,29 +1,23 @@
-# Channels
+# latest
 
-`fuelup` adopts a simplified version of `rustup` [channels](https://rust-lang.github.io/rustup/concepts/channels.html). Currently, only the `latest` channel will be published and serve as a source of distribution of Fuel toolchain binaries.
+### `latest`
```
Comment (kayagokalp) [inline docs/src/concepts/channels/latest.md:3]:
```suggestion
## `latest`
```

My `markdown` linter, complains about header levels not increased one by one Although I am not sure if this is visually acceptable 👀

File: docs/src/basics.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+fuelup toolchain install latest
+```
+
+## Keeping `fuelup` up to date
+
+You can request that `fuelup` update itself to the latest version of `fuelup`
+by running:
+
+```sh
+$ fuelup self update
```
Comment (kayagokalp) [inline docs/src/basics.md:29]:
```suggestion
fuelup self update
```

We are not using `$` in other examples such as:

```sh
 fuelup toolchain install latest
```

Also linter complains about the usage of `$`: 

```
MD014/commands-show-output: Dollar signs used before commands without showing output
```
[Cool reference](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md014---dollar-signs-used-before-commands-without-showing-output)

File: docs/src/concepts/toolchains.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+1. Official toolchains which track the official release [channels] (only _latest_ for now);
+2. Custom toolchains and install individual components in a modular manner.
+
+[channels]: channels/index.md
+
+## Toolchain specification
+
+Standard release channel toolchain names have the following form:
+
+```
```
Comment (kayagokalp) [inline docs/src/concepts/toolchains.md:15]: This is also caught by the linter with the following reference, I think this is not a language (not sure though 😄 ) maybe we can place a `text` next to it.

```
MD040/fenced-code-language: Fenced code blocks should have a language specified
```
[Cool reference 2](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md040---fenced-code-blocks-should-have-a-language-specified)

```suggestion
```text
```

File: docs/src/contributing_to_fuelup.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+If the pull request resolves the relevant issues, and you want GitHub to close these issues automatically after it merged into the default branch, you can use the syntax (`KEYWORD #ISSUE-NUMBER`) like this:
+
+```
+close #123
+```
+
+If the pull request links an issue but does not close it, you can use the keyword `ref` like this:
+
+```
```
Comment (kayagokalp) [inline docs/src/contributing_to_fuelup.md:83]:
```suggestion
```text
```

^

File: docs/src/contributing_to_fuelup.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ref #456
+```
+
+Multiple issues should use full syntax for each issue and separate by a comma, like:
+
+```
```
Comment (kayagokalp) [inline docs/src/contributing_to_fuelup.md:89]:
```suggestion
```text
```

^

**Review by kayagokalp — DISMISSED** [review]
Looks super! 🚀 I just left a couple of small nits w.r.t `markdown` linting. I am not sure if the things that my linter complained about are coming from a standard but I think as we build up the `fuelup` book from ground up would be cool to lint our markdowns. That being said this really looks cool and nothing I mentioned here is a blocker.

**Comment by bingcicle** [comment]
Thanks for the catches! @kayagokalp

**Review by eureka-cpu — COMMENTED** [review]
LGTM, deferring to @kayagokalp / @adlerjohn

File: docs/src/basics.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+fuelup toolchain install latest
+```
+
+## Keeping the Fuel toolchain up to date
+
+The Fuel toolchain is distributed on one [release channel]: latest (with nightly being a WIP).
+`fuelup` uses the `latest` channel by default, which
+represents the latest release of the Fuel toolchain.
+
+When new versions of the components within the Fuel toolchain is released,
```
Comment (JoshuaBatty) [inline docs/src/basics.md:16]: ~~is released~~ -> are released

**Review by JoshuaBatty — COMMENTED** [review]
Nice one @bingcicle! just one small nit.

Will wait for either @adlerjohn or @mitchmindtree to take a look for final approval.

**Comment by bingcicle** [comment]
I would like to make a v0.3.0 release but this doesn't necessarily have to go in together with the release itself - would ideally like either @adlerjohn or @mitchmindtree to vet the content to make sure everything's worded appropriately before merging and making this official.

Would like to go ahead with the release to test out custom toolchains in the wild asap! (esp in https://github.com/FuelLabs/sway-applications/issues/199)

File: docs/src/concepts/channels/index.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Channels
+
+`fuelup` adopts a simplified version of `rustup` [channels](https://rust-lang.github.io/rustup/concepts/channels.html). Currently, only the `latest` channel will be published and serve as a source of distribution of Fuel toolchain binaries.
+
+| Channel      | Source          | Integration Tested   | Update Frequency         | Available |
+| ------------ | --------------- | -------------------- | ------------------------ | --------- |
+| **[latest]** | published bins  | ✔️                   | checked every 30 minutes | ✔️        |
```
Comment (adlerjohn) [inline docs/src/concepts/channels/index.md:7]: Why is `latest` in square brackets? It's not a link?

File: docs/src/concepts/channels/latest.md [hunk 8]
```diff
@@ -1,29 +1,21 @@
...
-### `nightly`
-
-This is a WIP - we aim to support `nightly` channel soon. 
-
 ## Developer Guide
 
 ### Understanding the `latest` workflow
 
-_Note: Reading the information below is only really necessary if you wish to contribute to the workflows or want a deeper understanding on how channels are updated_
+_Note: Reading the information below is only really necessary if you wish to contribute to the workflows or want a deeper understanding on how channels are updated._
```
Comment (adlerjohn) [inline docs/src/concepts/channels/latest.md:18]: Consider

```suggestion
> **Note**
> Reading the information below is only really necessary if you wish to contribute to the workflows or want a deeper understanding on how channels are updated.
```

to look cute.

File: docs/src/concepts/channels/index.md (hunk 7 above)
Comment (bingcicle) [reply docs/src/concepts/channels/index.md:7]: Ah, it's meant to link to the latest channel on the next page but it was broken before.

https://github.com/FuelLabs/fuelup/pull/134/files#diff-b9e0f260d64132ce19cdd88e9a284bfbf76c2b2eed0056d9f5faca98bb4a82a8R12

 Thanks for the catch!

**Comment by adlerjohn** [comment]
Could you add a markdownlint CI workflow in a future PR? A la https://github.com/FuelLabs/fuel-specs/pull/397

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**